cryptography/src/commonMain/kotlin/com/wire/kalium/cryptography/utils/ChaCha20Utils.kt

```diff
@@ -0,0 +1,164 @@
+package com.wire.kalium.cryptography.utils
+
+import com.ionspin.kotlin.crypto.LibsodiumInitializer
+import com.ionspin.kotlin.crypto.pwhash.crypto_pwhash_SALTBYTES
+import com.ionspin.kotlin.crypto.secretstream.SecretStream
+import com.ionspin.kotlin.crypto.secretstream.crypto_secretstream_xchacha20poly1305_ABYTES
+import com.ionspin.kotlin.crypto.secretstream.crypto_secretstream_xchacha20poly1305_HEADERBYTES
+import com.ionspin.kotlin.crypto.secretstream.crypto_secretstream_xchacha20poly1305_TAG_FINAL
+import com.ionspin.kotlin.crypto.secretstream.crypto_secretstream_xchacha20poly1305_TAG_MESSAGE
+import com.wire.kalium.cryptography.backup.Backup
+import com.wire.kalium.cryptography.backup.Backup.Companion.BACKUP_FILE_HEADER_LENGTH
+import com.wire.kalium.cryptography.kaliumLogger
+import okio.Buffer
+import okio.Sink
+import okio.Source
+import okio.buffer
+import kotlin.random.Random
+import kotlin.random.nextUBytes
+
+@OptIn(ExperimentalUnsignedTypes::class)
+internal class ChaCha20Utils {
+
+    @Suppress("TooGenericExceptionCaught")
+    suspend fun encryptBackupFile(
+        backupDataSource: Source,
+        outputSink: Sink,
+        backup: Backup
+    ): Long {
+
+        initializeLibsodiumIfNeeded()
+        var encryptedDataSize = 0L
+        val additionalInformation: UByteArray = ubyteArrayOf()
+
+        try {
+            val chaCha20Key = backup.generateChaCha20Key()
+            val hashedUserId = backup.hashUserId()
```

Comment:
Why expose this a separate function? `backup.provideHeaderBuffer` couldn't calculate it internally?